src/network/serverpackethandler.cpp

```diff
@@ -969,8 +969,8 @@ bool Server::checkInteractDistance(RemotePlayer *player, const f32 d, const std:
 	else if (max_d < 0)
 		max_d = BS * 4.0f;
 
-	// cube diagonal: sqrt(3) = 1.732
-	if (d > max_d * 1.732) {
+	// Cube diagonal: sqrt(3) = 1.732
+	if (d > max_d + 1.732f * BS) {
```

Comment:
I don't think this is correct.

The idea here is that the distance is up to `sqrt(3)` **times** longer when measuring through the cube diagonal, as opposed to a straight line (which has a length of `d`).

Reply:
If the node uses a larger selection box, then this check adds another half node tolerance in terms of allowed interact distance. `max_d` and `d` are both the radius of the allowed interact area.

Reply:
> another

where does the cube diagonal tolerance go then?

> half node

`sqrt(3)` is the diagonal of a full node